src/ee/storage/persistenttable.cpp

```diff
@@ -189,6 +191,16 @@ PersistentTable::~PersistentTable() {
     }
 }
 
+void PersistentTable::checkContext(const std::string& operation) {
+   if ((isReplicatedTable() && !SynchronizedThreadLock::usingMpMemory()) ||
```

Comment:
check can be simplified as
```suggestion
   if (isReplicatedTable() != SynchronizedThreadLock::usingMpMemory()) {
```